packages/web/src/store/application/ui/buy-audio/sagas.ts

```diff
@@ -627,15 +641,40 @@ function* startBuyAudioFlow({
     yield* put(transferCompleted())
 
     // Update wallet balance optimistically
-    const outputAmount = weiToString(convertWAudioToWei(transferAmount))
+    const outputAmount = convertWAudioToWei(transferAmount)
     yield* put(
       increaseBalance({
-        amount: outputAmount
+        amount: weiToString(outputAmount)
+      })
+    )
+
+    // Record success
+    yield* put(
```

Comment:
Brutal conversions 😆 Maybe we should log these out also?

Reply:
I think we'll get the result from what's in Amplitude :)